hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ReadOnlyZKClient.java

```diff
@@ -258,6 +263,21 @@ public void closed(IOException e) {
     }
   }
 
+  private static TimerTask getTimerTask(final long timeoutMs, final CompletableFuture<?> future, final String api) {
+    return timeout -> {
+      if (!future.isCancelled() && !future.isDone() && !future.isCompletedExceptionally()) {
+        future.completeExceptionally(new DoNotRetryIOException( "Zookeeper " + api + " could not be completed in " + timeoutMs + " ms"));
+      }
+    };
+  }
+
+  public CompletableFuture<byte[]> getWithTimeout(final String path, final long timeoutMs, final HashedWheelTimer retryTimer) {
```

Comment:
I prefer we just pass the retryTimer in when constructing ReadOnlyZKClient and store it as a class field.